lib/oli_web/account_lookup_cache.ex

```diff
@@ -0,0 +1,134 @@
+defmodule OliWeb.AccountLookupCache do
+  @moduledoc """
+    Provides a cache that can be used for account lookups. This cache is backed by
+    Cachex for local storage and PubSub for remote distribution. Keys are set to expire
+    after 1 day in order to prevent stale data in our cache over a long time period.
+  """
+
+  use GenServer
+
+  alias Phoenix.PubSub
+
+  @cache_name :cache_account_lookup
+
+  # ----------------
+  # Client
+
+  def start_link(init_args),
+    do: GenServer.start_link(__MODULE__, init_args, name: __MODULE__)
+
+  def get(key),
+    do: GenServer.call(__MODULE__, {:get, key})
+
+  def delete(key),
+    do: GenServer.call(__MODULE__, {:delete, key})
+
+  def put(key, value),
+    do: GenServer.call(__MODULE__, {:put, key, value})
+
+  def update(key, value),
+    do: GenServer.call(__MODULE__, {:update, key, value})
+
+  # ----------------
+  # Server callbacks
+
+  def init(_) do
+    {:ok, _pid} = Cachex.start_link(@cache_name, stats: true)
+    PubSub.subscribe(Oli.PubSub, cache_topic())
+
+    {:ok, [], {:continue, :init}}
+  end
+
+  def handle_continue(:init, state) do
+    nodes = Node.list()
+
+    if length(nodes) > 0 do
+      # just pick a random node in the cluster to request the dump
+      send({__MODULE__, Enum.random(nodes)}, {:request_dump, self()})
+    end
+
+    {:noreply, state}
+  end
+
+  def handle_call({:get, key}, _from, state) do
+    {:reply, Cachex.get(@cache_name, key), state}
+  end
+
+  def handle_call({:delete, key}, _from, state) do
+    case Cachex.del(@cache_name, key) do
+      {:ok, true} ->
+        PubSub.broadcast(Oli.PubSub, cache_topic(), {:delete, key})
+        {:reply, :ok, state}
+
+      _ ->
+        {:reply, :error, state}
+    end
+  end
+
+  def handle_call({:put, key, value}, _from, state) do
+    ttl = :timer.hours(24)
+
+    case Cachex.put(@cache_name, key, value, ttl: ttl) do
+      {:ok, true} ->
+        PubSub.broadcast(Oli.PubSub, cache_topic(), {:put, key, value, ttl})
+        {:reply, :ok, state}
+
+      _ ->
+        {:reply, :error, state}
+    end
+  end
+
+  def handle_call({:update, key, value}, _from, state) do
+    case Cachex.update(@cache_name, key, value) do
```

Comment:
Don't we need to update the ttl here?

Reply:
I don't have a strong opinion about it but since it's fresh data, I thought it might be useful to update the ttl. We can leave it like this and update it later if we need to